segment.go

```diff
@@ -294,6 +311,83 @@ func (s *SegmentBase) loadFields() error {
 	return nil
 }
 
+func (s *SegmentBase) loadFieldsNew() error {
+	pos := s.sectionsIndexOffset
+
+	if pos == 0 {
+		// this is the case only for older file formats
+		return s.loadFields()
+	}
+
+	// read the number of fields
+	numFields, sz := binary.Uvarint(s.mem[pos : pos+binary.MaxVarintLen64])
+	pos += uint64(sz)
+
+	var fieldID uint64
+
+	for fieldID < numFields {
+		addr := binary.BigEndian.Uint64(s.mem[pos : pos+8])
+
+		fieldSectionMap := make(map[uint16]uint64)
+
+		err := s.loadFieldNew(uint16(fieldID), addr, fieldSectionMap)
+		if err != nil {
+			return err
+		}
+
+		s.fieldsSectionsMap = append(s.fieldsSectionsMap, fieldSectionMap)
+
+		fieldID++
+		pos += 8
+	}
+
+	return nil
+}
+
+func (s *SegmentBase) loadFieldNew(fieldID uint16, addr uint64,
+	fieldSectionMap map[uint16]uint64) error {
+	pos := addr
+	fieldNameLen, sz := binary.Uvarint(s.mem[pos : pos+binary.MaxVarintLen64])
+	pos += uint64(sz)
+
+	fieldName := string(s.mem[pos : pos+fieldNameLen])
+	s.fieldsInv = append(s.fieldsInv, fieldName)
+	s.fieldsMap[fieldName] = uint16(fieldID + 1)
+
+	pos += fieldNameLen
+
+	fieldNumSections, sz := binary.Uvarint(s.mem[pos : pos+binary.MaxVarintLen64])
+	pos += uint64(sz)
+
+	for sectionIdx := uint64(0); sectionIdx < fieldNumSections; sectionIdx++ {
+		// read section id
+		fieldSectionType := binary.BigEndian.Uint16(s.mem[pos : pos+2])
+		pos += 2
+		fieldSectionAddr := binary.BigEndian.Uint64(s.mem[pos : pos+8])
+		pos += 8
+		fieldSectionMap[fieldSectionType] = fieldSectionAddr
+		if fieldSectionType == sectionInvertedIndex {
```

Comment:
bro like here we have dictLocs as []int
where idx is fieldID and value is the offset to the invertedSection.
but problem is there when u have 2 sections. like dictLocs[2] will always have offset to inverted section. `Maybe u can make dictLocs as [][]int where dictLocs[2][1] will be address of sectionID 1 for fieldID 2`

Reply:
yeah, dictLocs maps a fieldID to the starting offset of the term dictionary. the slice is populated only for sectionInvertedIndex, so the slice won't have any other section's information.

Reply:
so for other section u use another field in segment struct? ok i will use new then thanks bro